inference-engine/src/mkldnn_plugin/nodes/mkldnn_space_to_depth_node.cpp

```diff
@@ -0,0 +1,395 @@
+            optimizedParams.src_block_dims[1] = srcBlockedDims[1];
+            optimizedParams.src_block_dims[lastIdx] = extraBlockSize;
+            optimizedParams.src_block_dims[lastIdx - 1] = blockStep;
+
+            order[1] = 1;
+            order[2] = lastIdx - 1;
+            order[lastIdx - nSpatialDims] = lastIdx;
+        }
+
+        for (size_t i = 0; i < nSpatialDims; i++) {
+            optimizedParams.src_block_dims[i * 2 + 2] = dstBlockedDims[i + 2];
+            optimizedParams.src_block_dims[i * 2 + 3] = blockSize;
+
+            order[i + orderShiftForBlocks] = i * 2 + 2;
+            order[i + orderShiftForDims] = i * 2 + 3;
+        }
+    } else if (getParentEdgeAt(0)->getMemory().GetDesc().isTailCFormat()) {
+        srcDims.push_back(srcDims[1]);
+        dstDims.push_back(dstDims[1]);
+        srcDims.erase(srcDims.begin() + 1);
+        dstDims.erase(dstDims.begin() + 1);
+
+        size_t shift = static_cast<size_t>(mode == DEPTH_FIRST) + nSpatialDims + 1;
+        order[mode == SpaceToDepthMode::DEPTH_FIRST ? nSpatialDims + 1 : lastIdx] = lastIdx;
+        optimizedParams.src_block_dims[lastIdx] = srcDims.back();
+
+        for (size_t i = 0; i < nSpatialDims; i++) {
+            optimizedParams.src_block_dims[i * 2 + 1] = dstDims[i + 1];
+            optimizedParams.src_block_dims[i * 2 + 2] = blockSize;
+
+            order[i + 1] = i * 2 + 1;
+            order[i + shift] = i * 2 + 2;
+        }
+    } else {
+        size_t shift = static_cast<size_t>(mode == DEPTH_FIRST) + 1;
+        order[mode == DEPTH_FIRST ? 1 : nSpatialDims + 1] = 1;
+        optimizedParams.src_block_dims[1] = srcDims[1];
+
+        for (size_t i = 0; i < nSpatialDims; i++) {
+            optimizedParams.src_block_dims[i * 2 + 2] = dstDims[i + 2];
+            optimizedParams.src_block_dims[i * 2 + 3] = blockSize;
+
+            order[i + nSpatialDims + 2] = i * 2 + 2;
+            order[i + shift] = i * 2 + 3;
+        }
+    }
+
+    optimizedParams.dst_block_dims.resize(nDims);
+    for (size_t i = 0; i < nDims; i++)
+        optimizedParams.dst_block_dims[i] = optimizedParams.src_block_dims[order[i]];
+
+    optimizedParams.src_block_order.resize(nDims);
+    optimizedParams.dst_block_order.resize(nDims);
+    for (size_t i = 0; i < nDims; i++) {
+        optimizedParams.src_block_order[i] = i;
+        optimizedParams.dst_block_order[i] = i;
+    }
+
+    optimizedParams.src_block_strides.resize(nDims);
+    optimizedParams.dst_block_strides.resize(nDims);
+    optimizedParams.src_block_strides[lastIdx] = 1;
+    optimizedParams.dst_block_strides[lastIdx] = 1;
+    for (int i = lastIdx - 1; i >= 0; i--) {
+        optimizedParams.src_block_strides[i] =
+                optimizedParams.src_block_strides[i + 1] * optimizedParams.src_block_dims[i + 1];
+        optimizedParams.dst_block_strides[i] =
+                optimizedParams.dst_block_strides[i + 1] * optimizedParams.dst_block_dims[i + 1];
+    }
+
+    prepareConfigParams();
+}
```

Comment:
Code on lines 121-256 seems cumbersome. I suggest doing two things:
1. Select sections of code into separate methods, for example, by layout type.
2. It seems to me that with minor changes it would be possible to combine the code from here with the code from method `MKLDNNDepthToSpaceNode::createPrimitive()`. Is it possible?

Reply:
Thank you for your suggestions!

I created new common class with the same  for `DepthToSpace` and `SpaceToDepth` sections of code methods. And I wrote new lambda function for creating `order` and new `src_block_dims`. 
I tried to move the same code selections into common methods and avoid a lot of conditions with layout type or layer type (`DepthToSpace` or `SpaceToDepth`)